app/discovery/discovery.go

```diff
@@ -198,6 +198,32 @@ func (s *Service) Match(srv, src string) (res Matches) {
 	return res
 }
 
+func findMatchingMappers(mappers map[string][]URLMapper, srvName string) []URLMapper {
+	for mapperServer, mapper := range mappers {
```

Comment:
you can likely just do map's lookup first to prevent loop on mappers for strict match